src/kdf.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include "mococrw/kdf.h"
+#include "mococrw/openssl_wrap.h"
+
+namespace mococrw {
+
+KeyDerivationFunction::~KeyDerivationFunction() = default;
+
+
+
+
+class PBKDF2::Impl
+{
+public:
+    Impl(openssl::DigestTypes hashFunction, uint32_t iterations):
+        _hashFunction(hashFunction), _iterations(iterations) {}
+
+    ~Impl() = default;
+
+    Impl(Impl&&) = default;
```

Comment:
We are also missing these in the other Impl class for X963…

Reply:
You are right. They are not necessary as the class members have an implicit copy and move constructor, so that the compiler is able to generate one for Impl.